frontend/routes/components/log/log.component.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ *  Copyright (C) 2017 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import { DateTime } from './../dateTime/dateTime.component';
+import HoverableUsername from './../hoverableUsername/hoverableUsername.container';
+import { Container, UserMessage, SystemMessage, Info, Screenshot, ScreenshotMessage } from './log.styles';
+
+interface IProps {
+	comment: string;
+	viewpoint: any;
+	created: number;
+	owner: string;
+	action: any;
+	companyName: string;
+	userName: string;
+	teamspace: string;
+}
+
+export class Log extends React.PureComponent<IProps, any> {
+	public renderScreenshot = (viewpoint, comment) => {
+		return (
+			<>
+				<Screenshot src={viewpoint.screenshotPath} withMessage={!!comment} />
+				{comment && <ScreenshotMessage>{comment}</ScreenshotMessage>}
+			</>
+		);
+	}
+
+	public renderMessage = (action, comment, created) => {
+		if (action) {
+			return (
+				<SystemMessage>{action.text} at <DateTime value={created} format="HH:mm DD MMM" /></SystemMessage>
+			);
+		}
+		if (comment) {
+			return <UserMessage>{comment}</UserMessage>;
+		}
+		return null;
+	}
+
+	public renderInfo = (action, owner, created, teamspace) => {
+		if (!action) {
+			return (
+				<Info>
+					{<HoverableUsername teamspace={teamspace} name={owner} />}
+					<DateTime value={created} format="HH:mm DD MMM" />
+				</Info>
+			);
+		}
+		return null;
+	}
+
+	public render() {
+		const { action, comment, created, viewpoint, owner, teamspace } = this.props;
+		return (
+			<Container>
+				{
+					viewpoint && viewpoint.screenshotPath
```

Comment:
Please, use here `renderWhenTrue` pattern
```suggestion
       	get isScreenshot() {
       		return this.props.viewpoint && this.props.viewpoint.screenshotPath;
       	}
       	get isAction() {
       		return Boolean(this.props.action);
       	}

       	// (...)
       	
       	public render () {
       	// (...)
 				{this.renderScreenshot(this.isScreenshot)}
 				{this.renderMessage(!this.isScreenshot)}
 				{this.renderInfo(!this.isAction)}
```